packages/block-editor/src/components/rich-text/use-format-types.js

```diff
@@ -71,16 +71,18 @@ export function useFormatTypes( {
 				return false;
 			}
 
+			// The "Footnote" format requires special handling due to its nested interactive content tag.
 			if (
 				withoutInteractiveFormatting &&
-				interactiveContentTags.has( tagName )
+				( name === 'core/footnote' ||
+					interactiveContentTags.has( tagName ) )
 			) {
 				return false;
 			}
 
 			return true;
 		} );
-	}, [ allFormatTypes, allowedFormats, interactiveContentTags ] );
+	}, [ allFormatTypes, allowedFormats, withoutInteractiveFormatting ] );
```

Comment:
Fixes the `useMemo` dependencies unrelated to the bug fix.

* The `interactiveContentTags` set won't change during the render.
* The `withoutInteractiveFormatting` is a boolean and used inside the memo function.